hbase-client/src/main/java/org/apache/hadoop/hbase/client/Scan.java

```diff
@@ -447,7 +450,16 @@ public Scan withStopRow(byte[] stopRow, boolean inclusive) {
    * after this method will yield undefined results.</b></p>
    * @param rowPrefix the prefix all rows must start with. (Set <i>null</i> to remove the filter.)
    * @return this
+   * @deprecated since 3.0.0. The scan result might be unexpected in some cases.
+   *   e.g. startRow : "112" and rowPrefixFilter : "11"
+   *   The Result of this scan might contains : "111"
+   *   This method implements the filter by setting startRow and stopRow,
+   *   but does not take care of the scene where startRow has been set.
+   *   Users can compose this logic externally through
+   *   {@link ClientUtil#calculateTheClosestNextRowKeyForPrefix(byte[])},
+   *   so there is no need to implement it inside.
```

Comment:
`ClientUtil` is IA.Private, hence not really recommended for direct use by downstreamers.

Reply:
> `ClientUtil` is IA.Private, hence not really recommended for direct use by downstreamers.

@huaxiangsun suggest me to give more detail, so i put this link. What i mean is just user can do this things by themself. Should i remove this link?

Reply:
I think we can remove just this statement: 
```
   *   Users can compose this logic externally through
   *   {@link ClientUtil#calculateTheClosestNextRowKeyForPrefix(byte[])},
   *   so there is no need to implement it inside.
```
because `ClientUtil` is IA.Private, hence we should not recommend it in Javadoc.
@huaxiangsun Thought?

Reply:
>    *   Users can compose this logic externally through
>   *   {@link ClientUtil#calculateTheClosestNextRowKeyForPrefix(byte[])},
>   *   so there is no need to implement it inside.

+1 for remove these comment.

Reply:
+1 to remove. Sorry, it is just I did not know the alternative way to archive the goal.

Reply:
Yeah, this is a tough one. Ideally we can provide the link in comment but the class here is IA.Private.